### PR TITLE
refactor(package): update module

### PR DESCRIPTION
**Pull request by aaharu**
* Update module following `module-template`
* Coverage reports successfully

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  testEnvironment: 'node',
+  collectCoverage: true,
+  collectCoverageFrom: [
+    'lib/**/*.js'
+  ],
+  moduleNameMapper: {
```
Comment (ricardogobbosouza) [inline jest.config.js]: No need `moduleNameMapper`

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    '^~/(.*)$': '<rootDir>/lib/$1',
+    '^~~$': '<rootDir>',
+    '^@@$': '<rootDir>',
+    '^@/(.*)$': '<rootDir>/lib/$1'
+  },
+  transform: {
```
Comment (ricardogobbosouza) [inline jest.config.js]: No need `transform`

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    '^.+\\.js$': 'babel-jest'
+  },
+  forceExit: true
```
Comment (ricardogobbosouza) [inline jest.config.js]: No need `forceExit`

File: test/fixture/less/nuxt.config.js [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
   rootDir: resolve(__dirname, '../../../'),
-  srcDir: resolve(__dirname),
+  buildDir: resolve(__dirname, '.nuxt'),
+  srcDir: __dirname,
   css: ['~assets/a.less'],
   render: {
     resourceHints: false
   },
-  modules: ['@@'],
+  modules: [mod],
```
Comment (ricardogobbosouza) [inline test/fixture/less/nuxt.config.js]: Use `require`
```js
modules: [
  { handler: require('../../../') }
]
```

File: test/fixture/less/nuxt.config.js [hunk 5]
```diff
@@ -1,13 +1,15 @@
+import mod from '@@'
```
Comment (ricardogobbosouza) [inline test/fixture/less/nuxt.config.js]: No need

File: test/fixture/sass/nuxt.config.js [hunk 6]
```diff
@@ -1,8 +1,10 @@
+import mod from '@@'
```
Comment (ricardogobbosouza) [inline test/fixture/sass/nuxt.config.js]: No need

File: test/fixture/sass/nuxt.config.js [hunk 7]
```diff
@@ -11,7 +13,7 @@ module.exports = {
     scss: ['@/assets/nested/index.scss', 'mathsass'],
     sass: ['@/assets/nested/index.sass']
   },
-  modules: ['@@'],
+  modules: [mod],
```
Comment (ricardogobbosouza) [inline test/fixture/sass/nuxt.config.js]: Use `require`
```js
modules: [
  { handler: require('../../../') }
]
```

File: test/fixture/stylus/nuxt.config.js [hunk 8]
```diff
@@ -1,16 +1,18 @@
+import mod from '@@'
```
Comment (ricardogobbosouza) [inline test/fixture/stylus/nuxt.config.js]: No need

File: test/fixture/stylus/nuxt.config.js [hunk 9]
```diff
@@ -1,16 +1,18 @@
...
+  srcDir: __dirname,
   css: ['~assets/a.styl'],
   render: {
     resourceHints: false
   },
   styleResources: {
     stylus: ['~assets/variables.styl']
   },
-  modules: ['@@'],
+  modules: [mod],
```
Comment (ricardogobbosouza) [inline test/fixture/stylus/nuxt.config.js]: Use `require`
```js
modules: [
  { handler: require('../../../') }
]
```

File: babel.config.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+  presets: [
+    [
+      '@babel/preset-env', {
+        targets: {
+          esmodules: true
+        }
+      }
+    ]
+  ]
+}
```
Comment (ricardogobbosouza) [inline babel.config.js]: Remove `babel.config.js`, no need

File: package.json [hunk 11]
```diff
@@ -21,15 +21,11 @@
...
   "scripts": {
     "dev": "nuxt build --config-file test/fixture/configs/default.js",
-    "lint": "eslint lib test",
-    "test": "yarn run lint && jest",
-    "release": "standard-version && git push --follow-tags && npm publish",
-    "commitlint": "commitlint -e $GIT_PARAMS",
+    "lint": "eslint --ext .js,.vue lib test",
+    "test": "yarn lint && jest",
+    "release": "yarn test && standard-version && git push --follow-tags && npm publish",
     "coverage": "codecov"
```
Comment (ricardogobbosouza) [inline package.json]: No need `coverage` script

File: package.json [hunk 12]
```diff
@@ -47,54 +43,33 @@
...
   "dependencies": {
     "consola": "^2.4.0",
     "glob-all": "^3.1.0",
     "sass-resources-loader": "^2.0.0"
   },
   "devDependencies": {
-    "@commitlint/cli": "^7.5.0",
-    "@commitlint/config-conventional": "^7.5.0",
+    "@babel/core": "latest",
+    "@babel/preset-env": "latest",
```
Comment (ricardogobbosouza) [inline package.json]: No need `@babel/core` and `@babel/preset-env`

File: package.json [hunk 13]
```diff
@@ -47,54 +43,33 @@
...
-    "eslint-plugin-promise": "^4.0.1",
-    "eslint-plugin-standard": "^4.0.0",
-    "eslint-plugin-vue": "^5.1.0",
-    "get-port": "^4.1.0",
-    "husky": "^1.3.1",
-    "jest": "^24.1.0",
-    "jsdom": "^13.2.0",
+    "@nuxtjs/eslint-config": "latest",
+    "babel-eslint": "latest",
+    "babel-jest": "latest",
```
Comment (ricardogobbosouza) [inline package.json]: No need `babel-eslint` and `babel-jest`

File: package.json [hunk 14]
```diff
@@ -47,54 +43,33 @@
...
     "sass-resources-loader": "^2.0.0"
   },
   "devDependencies": {
-    "@commitlint/cli": "^7.5.0",
-    "@commitlint/config-conventional": "^7.5.0",
+    "@babel/core": "latest",
+    "@babel/preset-env": "latest",
+    "@commitlint/cli": "latest",
+    "@commitlint/config-conventional": "latest",
     "@nuxt/config": "^2.8.1",
```
Comment (ricardogobbosouza) [inline package.json]: No need `@nuxt/config`

File: package.json [hunk 15]
```diff
@@ -47,54 +43,33 @@
...
     "mathsass": "^0.10.1",
     "node-sass": "^4.11.0",
-    "nuxt": "^2.4.3",
-    "nuxt-edge": "^2.4.1-25812008.a303ea89",
+    "nuxt-edge": "latest",
     "sass-loader": "^7.1.0",
-    "standard-version": "^4.4.0",
+    "standard-version": "latest",
     "stylus": "^0.54.5",
     "stylus-loader": "^3.0.2"
```
Comment (ricardogobbosouza) [inline package.json]: Try to update all devDependencies to `latest`

File: package.json [hunk 16]
```diff
@@ -47,54 +43,33 @@
...
-    "testEnvironment": "node",
-    "collectCoverage": true,
-    "coveragePathIgnorePatterns": [
-      "/node_modules/",
-      "/test/fixture"
-    ],
-    "forceExit": true
-  },
   "dependencies": {
     "consola": "^2.4.0",
```
Comment (ricardogobbosouza) [inline package.json]: Update `consola` to `^2.10.1`

File: package.json (hunk 14 above)
Comment (aaharu) [reply package.json]: It's used at `index.d.ts`

File: jest.config.js (hunk 2 above)
Comment (aaharu) [reply jest.config.js]: OK, but `lib/module.js` uses `export default function` (ES module). 
There is a need to change it to `module.exports = function` (CommonJS), if uses `{ handler: require('../../../') }`.

which is better?

Comment (ricardogobbosouza) [reply jest.config.js]: Use `CommonJS` to be more backwards compatible with nuxt

File: package.json (hunk 14 above)
Comment (ricardogobbosouza) [reply package.json]: Ok, just move to a `types` folder

File: package.json (hunk 13 above)
Comment (aaharu) [reply package.json]: if remove `babel-eslint`, `npm run lint` has errors.

```
/****/src/github.com/aaharu/style-resources-module/lib/module.js
  0:0  error  Parsing error: Cannot find module 'babel-eslint'
Require stack:
- /****/src/github.com/aaharu/style-resources-module/node_modules/vue-eslint-parser/index.js
- /****/src/github.com/aaharu/style-resources-module/node_modules/eslint-plugin-vue/lib/utils/index.js
- /****/src/github.com/aaharu/style-resources-module/node_modules/eslint-plugin-vue/lib/rules/array-bracket-spacing.js
- /****/src/github.com/aaharu/style-resources-module/node_modules/eslint-plugin-vue/lib/index.js
- /****/src/github.com/aaharu/style-resources-module/node_modules/eslint/lib/cli-engine/config-array-factory.js
- /****/src/github.com/aaharu/style-resources-module/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js
- /****/src/github.com/aaharu/style-resources-module/node_modules/eslint/lib/cli-engine/cli-engine.js
- /****/src/github.com/aaharu/style-resources-module/node_modules/eslint/lib/cli-engine/index.js
- /****/src/github.com/aaharu/style-resources-module/node_modules/eslint/lib/cli.js
- /****/src/github.com/aaharu/style-resources-module/node_modules/eslint/bin/eslint.js

```

Comment (ricardogobbosouza) [reply package.json]: You need to update `.eslintrc.js` to
```js
module.exports = {
  root: true,
  extends: [
    '@nuxtjs'
  ]
}
```

File: .travis.yml [hunk 17]
```diff
@@ -11,4 +11,4 @@ install:
 script:
 - yarn test
 after_success:
-- yarn coverage
+- yarn run codecov
```
Comment (ricardogobbosouza) [inline .travis.yml]: Move to circleci

File: package.json [hunk 18]
```diff
@@ -47,54 +43,28 @@
...
-      "/node_modules/",
-      "/test/fixture"
-    ],
-    "forceExit": true
-  },
   "dependencies": {
-    "consola": "^2.4.0",
+    "consola": "^2.10.1",
     "glob-all": "^3.1.0",
     "sass-resources-loader": "^2.0.0"
```
Comment (ricardogobbosouza) [inline package.json]: Add `@nuxt/config` for types

File: lib/module.js [hunk 19]
```diff
@@ -2,7 +2,7 @@ const glob = require('glob-all')
 const consola = require('consola')
 const logger = consola.withScope('nuxt-style-resources')
 
-export default function nuxtStyledResources() {
+module.exports = function nuxtStyledResources () {
```
Comment (ricardogobbosouza) [inline lib/module.js]: No need function name

File: .travis.yml (hunk 17 above)
Comment (ricardogobbosouza) [reply .travis.yml]: Use https://github.com/nuxt-community/module-template/blob/master/template/.circleci/config.yml

Comment (aaharu) [reply .travis.yml]: Could you do that work by any one of project members?
I can't set Codecov's token to CircleCI.

Comment (ricardogobbosouza) [reply .travis.yml]: We can ask @manniL

**Comment by rchl** [comment]
Thanks for this PR.

Some of those changes are already on master and the remaining ones that still looked relevant I've pulled in to https://github.com/nuxt-community/style-resources-module/pull/166

**Pull request closed without merging**